pkg/chunk/cache/cache.go

```diff
@@ -7,8 +7,8 @@ import (
 
 // Cache byte arrays by key.
 type Cache interface {
-	Store(ctx context.Context, key string, buf []byte) error
-	Fetch(ctx context.Context, keys []string) (found []string, bufs [][]byte, missing []string, err error)
+	Store(ctx context.Context, key []string, buf [][]byte)
```

Comment:
Having to constantly keep track of two arrays is not my favorite pattern, it might be nicer to have a small struct with key and buf in it. WDYT?

Reply:
I tried with a struct, but I kinda prefer this way.  Mainly because there are a few places where we have to convert the list of bufs into a list of chunks (or index entries, or query results) and not having to have N different types for that was helpful.  But I don't feel strongly, so if you do I'll happily try it out!

Reply:
I also do not feel strongly, just wanted to have a quick discussion. 👍